src/libraries/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Directory.Changed.cs

```diff
@@ -44,57 +44,59 @@ public void FileSystemWatcher_Directory_Changed_WatchedFolder()
         [InlineData(true)]
         public void FileSystemWatcher_Directory_Changed_Nested(bool includeSubdirectories)
         {
-            using (var dir = new TempDirectory(GetTestFilePath()))
-            using (var firstDir = new TempDirectory(Path.Combine(dir.Path, "dir1")))
-            using (var nestedDir = new TempDirectory(Path.Combine(firstDir.Path, "nested")))
-            using (var watcher = new FileSystemWatcher(dir.Path, "*"))
+            string firstDir = Path.Combine(TestDirectory, "dir1");
+            string nestedDir = Path.Combine(firstDir, "nested");
+            Directory.CreateDirectory(nestedDir);
+
+            using (var watcher = new FileSystemWatcher(TestDirectory, "*"))
             {
                 watcher.IncludeSubdirectories = includeSubdirectories;
                 watcher.NotifyFilter = NotifyFilters.DirectoryName | NotifyFilters.Attributes;
 
-                var attributes = File.GetAttributes(nestedDir.Path);
-                Action action = () => File.SetAttributes(nestedDir.Path, attributes | FileAttributes.ReadOnly);
-                Action cleanup = () => File.SetAttributes(nestedDir.Path, attributes);
+                var attributes = File.GetAttributes(nestedDir);
+                Action action = () => File.SetAttributes(nestedDir, attributes | FileAttributes.ReadOnly);
+                Action cleanup = () => File.SetAttributes(nestedDir, attributes);
 
                 WatcherChangeTypes expected = includeSubdirectories ? WatcherChangeTypes.Changed : 0;
-                ExpectEvent(watcher, expected, action, cleanup, nestedDir.Path);
+                ExpectEvent(watcher, expected, action, cleanup, nestedDir);
             }
         }
 
         [ConditionalFact(typeof(MountHelper), nameof(MountHelper.CanCreateSymbolicLinks))]
         public void FileSystemWatcher_Directory_Changed_SymLink()
         {
-            using (var testDirectory = new TempDirectory(GetTestFilePath()))
-            using (var dir = new TempDirectory(Path.Combine(testDirectory.Path, "dir")))
-            using (var tempDir = new TempDirectory(Path.Combine(testDirectory.Path, "tempDir")))
-            using (var file = new TempFile(Path.Combine(tempDir.Path, "test")))
-            using (var watcher = new FileSystemWatcher(dir.Path, "*"))
+            string dir = Path.Combine(TestDirectory, "dir");
+            string tempDir = Path.Combine(dir, "tempDir");
+            string file = Path.Combine(tempDir, "test");
+            Directory.CreateDirectory(tempDir);
```

Comment:
```suggestion
            Directory.CreateDirectory(Path.Combine(dir, "tempDir"));
            string file = Path.Combine(dir, "tempDir", "test");
            
```
is that cleaner? up to you